evidence.go

```diff
@@ -78,10 +78,16 @@ func (e *Evidence) FromCOSE(cwt []byte, supportedProfiles ...string) error {
 	// the condition has been checked at the top of the function by the call
 	// to checkSupportedProfiles
 	for _, profile := range supportedProfiles {
-		err = e.Claims.validate(profile)
-		if err == nil {
+		e := e.Claims.validate(profile)
+		if e == nil {
```

Comment:
Another question, if you could please help me clarify:
The intent of line 80 is to check valid claims for a range of profiles.
However as per line 82, one one would break on the first opportunity of a successful validate ?
The two logics are counter-intuitive, isn't it?

Reply:
as discussed: one needs to run the validation against each of the supported profiles to see if we can get a match.  The first profile that match is a win an we break out.